lib/ecstatic.js

```diff
@@ -330,6 +339,19 @@ var ecstatic = module.exports = function (dir, options) {
 ecstatic.version = version;
 ecstatic.showDir = showDir;
 
+function hasGzipId12(gzipped, cb) {
+  var stream = fs.createReadStream(gzipped, { start: 0, end: 1, autoClose: true }),
+      isGzip = false;
+
+  stream.on("close", function () {
+    cb(isGzip);
+  });
+
+  stream.on("data", function (chunk) {
```

Comment:
Right, but my concern isn't short files, it's long files.

(I'll preface this by saying that I did some research on whether readFile supports start and end; unfortunately it doesn't. So we can't do that.)

My concern is this: I don't think streams guarantee any minimum length on data buffers, meaning it's possible to get a data event with just the first byte but not the second.

This can be solved with https://www.npmjs.com/package/concat-stream pretty easily, and also help us avoid having to listen for stream events directly which isn't the funnest.